spec/jobs/irs_attempts_events_batch_job_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe IrsAttemptsEventsBatchJob, type: :job do
+  describe '#perform' do
+    events = { key1: 'SomeEncryptedEvent1' }
+    Result = Struct.new(:filename, :iv, :encrypted_key, :encrypted_data, keyword_init: true)
+
+    let(:redis_client) { instance_double(IrsAttemptsApi::RedisClient) }
+    let(:envelope_encryptor) { instance_double(IrsAttemptsApi::EnvelopeEncryptor) }
+
+    let(:file_double) { instance_double(File) }
+
+    context 'IRS attempts API is enabled' do
+      before do
+        allow(IdentityConfig.store).to receive(:irs_attempt_api_enabled).and_return(true)
+
+        allow(IrsAttemptsApi::RedisClient).
+          to receive(:new).
+          and_return(redis_client)
+
+        allow(redis_client).
+          to receive(:read_events).
+          and_return(events)
+
+        allow(IrsAttemptsApi::EnvelopeEncryptor).
+          to receive(:encrypt).
+          and_return(
+            Result.new(
+              filename: 'test_filename',
+              encrypted_data: 'EnvelopeEncryptedEvents',
+            ),
+          )
+
+        allow(File).
+          to receive(:exists?).
+          and_return(true)
+
+        allow(File).
+          to receive(:open).
+          and_return(file_double)
+
+        allow(file_double).
+          to receive(:write)
+
+        allow(file_double).
+          to receive(:close)
+
+        allow(file_double).
+          to receive(:path).
+          and_return('./attempts_api_output/test_filename')
```

Comment:
let's not stub the filesystem either

if we *need* we could use [fakefs](https://github.com/fakefs/fakefs) but I think we're better of using `Tempfile`

Reply:
This was my fault. 👼 I suggested this as an alternative to spamming log files with every test run.

@Rwolfe-Nava Zach just linked me to https://github.com/18F/identity-idp/blob/82843ecc19a00fb3d5b5fa69713908ea24ffa9f1/spec/services/data_requests/write_cloudwatch_logs_spec.rb#L32-L37 in talking about this, which creates a temp directory and then cleans up at the end of the test. Could be an alternative.

Reply:
Done